substrafl/model_loading.py

```diff
@@ -147,93 +144,191 @@ def _get_train_task_from_round(client: substra.Client, compute_plan_key: str, ro
 
-    if len(local_train_tasks) == 0:
-        raise TrainTaskNotFoundError(
-            f"The given compute plan `{compute_plan_key}` has no train task of round {round_idx} "
+    if len(local_tagged_tasks) == 0:
+        raise exceptions.TaskNotFoundError(
+            f"The given compute plan `{compute_plan_key}` has no {tag} task of round {round_idx} "
             f"hosted on the organization {org_id}"
         )
 
-    elif len(local_train_tasks) > 1:
-        raise MultipleTrainTaskError(
-            "The given compute plan has {} local train tasks of round_idx {}. Downloading a model "
-            "from an experiment containing multiple TrainDataNodes hosted on the same organization is "
-            "not supported yet in local mode.".format(
-                str(len(local_train_tasks)),
-                str(round_idx),
-            )
+    elif len(local_tagged_tasks) > 1:
+        raise exceptions.MultipleTaskError(
+            f"The given compute plan has {len(local_tagged_tasks)} local {tag} tasks of round {round_idx}. "
         )
-    local_train_task = local_train_tasks[0]
+    local_task = local_tagged_tasks[0]
+    return local_task
+
+
+def _get_task_from_rank(
+    client: substra.Client, compute_plan_key: str, tag: str, rank_idx: Optional[int] = None
+) -> substra.models.Task:
+    """Return the task:
+
+        - hosted on the given client organization
+        - belonging to the given compute plan
+        - of the given rank_idx
+
+    Args:
+        client (substra.Client): Substra client where to fetch the task from.
+        compute_plan_key (str): Compute plan key to fetch the task from.
+        rank_idx (int): Rank of the strategy to fetch the task from. If set to None,
+            the last ending task will be considered. (Default to None)
+        tag (str): Tag of the task to get.
 
-    return local_train_task
+    Returns:
+        substra.models.Task: The task matching the given requirements.
+    """
+    org_id = client.organization_info().organization_id
 
+    if rank_idx is not None:
+        filters = {"compute_plan_key": [compute_plan_key], "worker": [org_id], "rank": [rank_idx]}
+        local_tasks = client.list_task(filters=filters)
+        local_tagged_tasks = [t for t in local_tasks if t.tag == tag]
 
-def _load_algo(algo_path: Path, extraction_folder: Path) -> Any:
-    """Load into memory a serialized (and compressed (.tar.gz)) substrafl algo within the given algo_path.
+    else:
+        filters = {"compute_plan_key": [compute_plan_key], "worker": [org_id]}
+        local_tasks = client.list_task(filters=filters, order_by="end_date")
+        local_tagged_tasks = [t for t in local_tasks if t.tag == tag]
+
+    if len(local_tagged_tasks) == 0:
+        raise exceptions.TaskNotFoundError(
+            f"The given compute plan `{compute_plan_key}` has no {tag} task of rank {rank_idx} "
+            f"hosted on the organization {org_id}"
+        )
+
+    elif len(local_tagged_tasks) > 1:
```

Comment:
If `rank_idx` is None, shouldn't we return the last task, in case of multiples, rather than raising? 
At least, that would be the behaviour I expect reading the docstring 🤔

Reply:
Nice catch thanks a lot !